riscv/src/compiler.rs

```diff
@@ -841,6 +875,17 @@ fn only_if_no_write_to_zero_vec(statements: Vec<String>, reg: Register) -> Vec<S
     }
 }
 
+// let coprocessor_substitutions = vec![("poseidon_coprocessor", "")];
+static COPROCESSOR_SUBSTITUTIONS: &'static [(&'static str, &'static str)] =
+    &[("poseidon_coprocessor", "x10 <=X= poseidon(x10, x11);")];
+
+fn has_coprocessor_substitution(label: &str) -> Option<&'static str> {
```

Comment:
A better name would be `try_to_coprocessor_substitution`. `has_*` should return a boolean.

Reply:
I think so yes!